app/Models/User.ts

```diff
@@ -24,15 +35,18 @@ export default class User extends BaseModel {
   @column.dateTime({ autoCreate: true, autoUpdate: true })
   public updatedAt: DateTime
 
-  @column({
-    prepare: (value: string[]): string => `{${value.join(',')}}`,
-    consume: (value: string): string[] => value.slice(1, -1).split(','),
-  })
+  @enumColumn()
```

Comment:
it would be better to use a SQL set / enum instead of a custom type

Reply:
Not sure to understand what you mean, we use an sql array of enums so this seems fine to me. I just think the name of the decorator should be `enumArrayColumn` instead I realize

Reply:
I think the enum serialization is already handled in the Adonis ORM
And have a look at this resource : https://docs.adonisjs.com/reference/database/table-builder#enum--enu
It might be easier to understood if you use the build-in methods instead of raw SQL

Reply:
I can't use an enum there because it's an array of enum that I need. So someone can have "a" and "b", and in sql it's stored as `{a,b}`. Does it make sense?